### PR TITLE
Close the help when clicking on the main content

### PR DESCRIPTION
**Pull request by cquiroz**
This is a small improvement to the help system. Now you can click outside the help to close it

### CODE REVIEW

File: explore/src/main/scala/explore/ExploreLayout.scala [hunk 1]
```diff
@@ -68,6 +75,9 @@ object ExploreLayout {
                 props.r.renderP(props.view)
               )
             )
+          )(
+            ^.onClick -->
+              helpView.set(none).runAsyncAndForgetCB.when_($.state.helpVisible)
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/ExploreLayout.scala:80]: What happens if we always set `none` here without tracking the status in the state?

Comment (cquiroz) [reply explore/src/main/scala/explore/ExploreLayout.scala:80]: Actually we should be able to simplify. I got a bit confused by this one getting called twice but it was avoided by preventing propagation on the help icon

File: explore/src/main/scala/explore/ExploreLayout.scala [hunk 2]
```diff
@@ -76,10 +86,11 @@ object ExploreLayout {
   private val component =
     ScalaComponent
       .builder[Props]
-      .stateless
-      .render_P { p =>
-        IfLogged(p.view)((renderFn _).reusable(p))
+      .initialState(State(false))
+      .renderP { ($, p) =>
+        IfLogged(p.view)((renderFn($) _).reusable(p))
```
Comment (rpiaggio) [inline explore/src/main/scala/explore/ExploreLayout.scala:91]: I'm not sure I would recommend this pattern of using `$` to build a reusable function. In this case the state is used in the function, but the state's reusability is not taken into account when computing the reusability of the function.

In other places I've added a `View[State]` as a regular parameter to `renderFn`, and then you can pass a `ViewF.fromState[IO]($)`.

**Review by rpiaggio — APPROVED** [review]
LGTM

**Comment by cquiroz** [comment]
This was simplified a lot taking the State out of `ExploreLayout`

**Pull request merged**